examples/ngsi_v2/e11_ngsi_v2_context_specific_models.py

```diff
@@ -0,0 +1,151 @@
+
+from filip.models.ngsi_v2.context import ContextEntity, ContextAttribute
+
+
+# Define a context specific attribute model, e.g. an attribute that holds
+# the postal address of location.
+
+
+# First, define the model for the address structure itself. Here, we use the
+# schema.org definition for a postal address.
+class AddressModel(BaseModel):
+    """
+    https://schema.org/PostalAddress
+    """
+
+    model_config = ConfigDict(populate_by_name=True)
+
+    address_country: str = Field(
+        alias="addressCountry",
+        description="County code according to ISO 3166-1-alpha-2",
+    )
+    address_region: str = Field(
+        alias="addressRegion",
+        default=None,
+    )
+    address_locality: str = Field(
+        alias="addressLocality",
+        description="The locality in which the street address is, and which is "
+        "in the region. For example, Mountain View.",
+    )
+    postal_code: str = Field(
+        alias="postalCode",
+        default=None,
+        description="The postal code. For example, 94043.",
+    )
+    street_address: str = Field(
+        alias="streetAddress",
+        default=None,
+        description="The street address. For example, 1600 Amphitheatre Pkwy.",
+    )
+
+
+# Now, define a context specific attribute for the address. Here, we use the
+# previously defined address model as type for the value.
+# Furthermore, we set the default type of the attribute to "PostalAddress"
+# to give the user a type-hint. Your may also allow multiple types during
+# creation.
+class AddressAttribute(ContextAttribute):
+    """
+    Context attribute for address
+    """
+
+    def __init__(self, type: str = None, **data):
+        if type is None and self.model_fields["type"].default:
+            type = self.model_fields["type"].default
+        super().__init__(type=type, **data)
+
+    # add default for type if not explicitly set
+    type: Literal["PostalAddress"] = FieldInfo.merge_field_infos(
+        # First position is the field info of the parent class
+        ContextAttribute.model_fields["type"],
+        # set the default value
+        default="PostalAddress",
+        # overwrite the description
+        description="Type of the Address",
+        # freeze the field if necessary, e.g. if you want to prevent the user
+        # from changing the type after creation.
+        frozen=True,
+        # for more options see the pydantic documentation
+    )
+    # add the value model. If you want to enforce the value to be set you can
+    # remove the default value
+    value: AddressModel = None
+
+
+# Define a context specific model for a weather station
+# First, inherit from the ContextEntity class, where each additional attribute
+# must be of type ContextAttribute
+class WeatherStation(ContextEntity):
+    """
+    A context specific model for a weather station
+    """
+    # add default for type if not explicitly set
+    type: str = FieldInfo.merge_field_infos(
```

Comment:
What is the difference here to simply use the following code?
```python
    type: Literal["PostalAddress"] = Field(
        # First position is the field info of the parent class
        # set the default value
        default="PostalAddress",
        # overwrite the description
        description="Type of the Address",
        # freeze the field if necessary, e.g. if you want to prevent the user
        # from changing the type after creation.
        frozen=True,
        # for more options see the pydantic documentation
    )
```

Reply:
No, it has to be mger_field because otherwise you would overwrite the complete field info from the parent Field, e.g. also patterns etc.